src/libraries/System.Net.Quic/src/System/Net/Quic/Interop/msquic.cs

```diff
@@ -21,6 +21,7 @@ namespace Microsoft.Quic
     internal unsafe partial struct QUIC_BUFFER
     {
         public Span<byte> Span => new(Buffer, (int)Length);
+        public readonly ReadOnlySpan<byte> ReadOnlySpan => new(Buffer, (int)Length);
```

Comment:
If Buffer doesn't refer to memory in the struct / same allocation itself (which it doesn't seem to to me), the existing `Span` member could actually just be marked `readonly` instead of adding a new member - if it does refer to memory within the same allocation / struct, then they should both be marked `[UnscopedRef]`.

Reply:
It refers to native memory (Buffer is void*).

I was unsure whether to make the Span property itself readonly instead. I don't really like being able to mutate data through readonly QUIC_BUFFER instance (hence explicit ReadOnlySpan property), but fact is that the possibility is not going away and keeping it like this allows the defensive copy problem to pop up elsewhere, so I will probably just make the Span readonly